### PR TITLE
docs(how-to guides): added documentation for maps creation

### PR DESCRIPTION
**Pull request by n-patiphon**
Signed-off-by: Patiphon Narksri <n.patiphon@gmail.com>

## Description
This PR adds a how-to page describing map creation. 

The documentation covers both specifications and commonly used tools to create point cloud and vector maps. Moreover, it includes a list of companies that currently provide Autoware-compatible map creation services.

This PR directly addresses #32.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by n-patiphon** [comment]
Hi, @mitsudome-r. I could not assign you as a reviewer. Please allow me to notify you here.

**Comment by n-patiphon** [comment]
Hi, @shmpwk the PR is now open.

**Review by shmpwk — COMMENTED** [review]
Could you add headline here?
https://github.com/autowarefoundation/autoware-documentation/blob/9054824ba51f0f436faf80aefb57bd7b053b854b/docs/how-to-guides/index.md

**Comment by n-patiphon** [comment]
> Could you add headline here? https://github.com/autowarefoundation/autoware-documentation/blob/9054824ba51f0f436faf80aefb57bd7b053b854b/docs/how-to-guides/index.md

@shmpwk Added in 2086c0a

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (shmpwk) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:29]: In order to get vector map, the best solution might be using vector map builder. 
IMO, it would be nice to get that answer a little sooner, since at the moment that information is at the end of the section.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (shmpwk) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:31]: I'm not sure but the word "vector map" and "road net map" sound like a different meaning. The latter is more rough map? 
IMO, it should unify the word  "vector map" if "road net map" is not commonly used.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:29]:
```suggestion
### Creating a vector map
```

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:9]:
```suggestion
### Creating a point cloud map
```

**Review by LalithVipulananthan — COMMENTED** [review]
Initial review, but will review again once @shmpwk's comments have been resolved.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (kminoda) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:21]: Should the map follow MGRS when using it for Autoware?

I think this is necessary when the readers want to use MGRS-based GNSS sensor input, but otherwise, IMO the map does not necessarily should follow the MGRS format. (correct me if I am wrong!)

So what do you think about removing this line? 
```suggestion
```

**Review by kminoda — COMMENTED** [review]
Overall, I think that this page is very descriptive and helpful for readers.
I left a few minor comments below:

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:27]: The Japanese version is [here](https://www.map4.jp/technology#mapiv_engine), but I cannot seem to find an English version

**Comment by n-patiphon** [comment]
Hi, all. Thank you very much for the comments. I will check them, respond, and apply revisions as soon as possible.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:25]: Oh, I didn’t notice the ros2 branch. I will mention this in the text too. Also, thanks for the discussion link!

Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:25]: I think that’s because there are more ROS1-based SLAM implementations that are known to work well and do not strictly require other sensors except LiDAR (at least for now). Also, the ones that use ROS2 don’t seem to be as mature as those that are based on ROS1.

However, I understand that many users may prefer an easy-to-use implementation just to get started quickly. I think maybe if I switch their position and talk about ROS2-based ones first that would help. Essentially, what @LalithVipulananthan  suggested.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:29]: The reason I didn’t mention VMB first is that I thought it was not free for the public (I’ve always used it with a TeirIV e-mail). I felt like I should try to cover free and open-source options as much as possible before mentioning any proprietary tools. 

Since it’s free as long (as you create an account with a valid e-mail), I will switch the order and mention VMB first.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:31]: I think “road network map” is actually a more common term compared to “vector map” when people refer to a high-definition map that has details of lanes, road boundaries, junctions, etc.

As far as I understand, “vector map” is a format name. It is a proprietary format from AISAN used in [Autoware.ai](http://autoware.ai/) (ROS1 Autoware) to represent a road network map. In the current Autoware (ROS2 Autoware), the chosen format is Lanelet2. 

I think the term “vector map” is equivalent to “Lanelet2”, “OpenDrive”, or “OpenStreetMap”. They are all formats used to describe a road network.

Because of this, I thought it would be more clear to just use both words. However, if it seems to cause confusion here, maybe I can modify the text a little bit to make it clearer.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:7]: According to this comment as well as [this](https://github.com/autowarefoundation/autoware-documentation/pull/140#discussion_r894219185) and [this](https://github.com/autowarefoundation/autoware-documentation/pull/140#discussion_r894224085), I think I will rename the title and just remove the first section. I also thought it was a bit weird too but so I'm gladly changing this.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:11]: Noted!

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:48]: I think both of them do ( at least Map IV does), but let me check again.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:31]: As I [replied](https://github.com/autowarefoundation/autoware-documentation/pull/140#discussion_r895357538) to the comment above, I think a road network map is a broader term and vector map actually refers to a format.

However, I think it's reasonable to keep the terminology consistent within the documentation (it is specifically aimed for Autoware users to understand after all). I think adding a sentence saying how vector map and road network map are related is a good idea.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:27]: It's not outdated but rather not ready yet. I requested the web team to create a landing page with that URL but it seems like it's not available just yet.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:21]: Basically, if GNSS is not used it can be in local coordinates as mentioned in the prior bullet point. However, if users want to georeference the map it has to be in MGRS. I think it's important to note that because there are other projections that can be used, UTM, Rectangular plane. Especially, since Autoware.ai used to use Rectangular plane projection and now we switched to MGRS.

Therefore, if you're ok with it I would like to keep this as some users might find it useful.

Comment (kminoda) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:21]: Thanks.

I assumed that we can change the format of map coordinates (among MGRS, UTM, or plane) from [this parameter in `gnss_poser`](https://github.com/autowarefoundation/autoware.universe/blob/f807914b60e70fb1ad4a50bdd942bd2330518fba/sensing/gnss_poser/launch/gnss_poser.launch.xml#L15) when GNSS is available. 
If my understanding is correct, how about something like this?

```suggestion
- When GNSS is available, the coordination system of the map should be compatible with the GNSS data. In the current Autoware Universe, we support MGRS, UTM, and Rectangular plane format for the map (see [the launch file of gnss_pose](https://github.com/autowarefoundation/autoware.universe/blob/main/sensing/gnss_poser/launch/gnss_poser.launch.xml#L15) to set the parameter).
```

Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:21]: I see. I think it's reasonable to include other coordinate systems as well. However, it also seems Lanelet2 loader also supports UTM actually (though it's not the [default choice](https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp#L53)). 

I will try to reword the sentence a little bit but will reflect your idea of including other coordinate systems. Thank you!

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:47]: - Since MapToolbox seems to be the better OSS option, I think we should mention that first and put JOSM at the bottom.
- I also simplified the description of JOSM, as I do not think we need to go into details about why JOSM is bad.

```suggestion
For open-source software options, [MapToolbox](https://github.com/autocore-ai/MapToolbox) is a plugin for [Unity](https://unity.com/) specifically designed to create Lanelet2 maps for Autoware.

Although [JOSM](https://josm.openstreetmap.de/) is another open-source tool that can be used to create Lanelet2 maps, be aware that a number of modifications must be done manually to make the map compatible with Autoware. This process can be tedious and time-consuming and so use of JOSM is not recommended.
```

**Review by LalithVipulananthan — COMMENTED** [review]
@n-patiphon Apologies for the extensive second review, but I went through the page again and noted some additional areas for potential improvement.

**Comment by n-patiphon** [comment]
@LalithVipulananthan No worries. Thank you very much for your additional comments! I will check them once I resolve all the initial suggestions.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:4]: I think the revised sentences look good to me. I'll add this change.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md:11]: @LalithVipulananthan Just to confirm. Did you mean to use H3 here?

**Review by LalithVipulananthan — APPROVED** [review]
LGTM! If there are any minor fixes or changes, we can always apply them in a future PR.

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md]: @LalithVipulananthan My apologies. I modified this section quite heavily before I saw this suggestion. Could you please take a look at the current version again and see if it still needs a modification?

**Review by shmpwk — APPROVED** [review]
Nice! Thank you for your improving!

**Comment by n-patiphon** [comment]
It seems like there are two checks that still fail. One of them is a dead link (https://tools.tier4.jp/vector_map_builder/). The other one is an unknown word (AISAN and AutoStream). However, the dead link is actually accessible after login and these words are specific names. 

Do they need to be addressed before this PR gets merged?

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:54]: Replaced the VMB link with one that works without logging into a T4 account.

```suggestion
The easiest way to create an Autoware-compatible vector map is to use [Vector Map Builder](https://tools.tier4.jp/feature/vector_map_builder_ll2/), a free web-based tool provided by [TIER IV, Inc.](https://www.tier4.jp/).
```

**Comment by LalithVipulananthan** [comment]
@kenji-miyake 
- I think that the "dead" link should be fixed, so I have updated the link with one that works without being logged in to a T4 account.
- However I am not sure what to do with the failing spell check: 
  - We can ignore the spell check
  - We can add `AISAN` and `Autostream`
  - We can add code to force the spellchecker to ignore those two words in this file
- Which one is best? 🤔

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (n-patiphon) [reply docs/how-to-guides/how-to-create-maps-for-autoware.md]: Thank you for the suggestion!

**Comment by kenji-miyake** [comment]
> We can add AISAN and Autostream

@LalithVipulananthan I prefer this option because it's often used in Autoware. But I think `Autostream` should be `AutoStream`, then I believe there is no spell-check error.
https://www.tomtom.com/products/autostream/

**Review by LalithVipulananthan — APPROVED** [review]
Added "AISAN" to the spellchecker word list, so this should be good to go!

File: docs/how-to-guides/how-to-create-maps-for-autoware.md
Comment (LalithVipulananthan) [inline docs/how-to-guides/how-to-create-maps-for-autoware.md:79]:
```suggestion
    The open-source [AutoStreamForAutoware tool](https://github.com/tomtom-international/AutoStreamForAutoware) can be used to convert an AutoStream map to a Lanelet2 map.
```

**Review by LalithVipulananthan — APPROVED** [review]
LGTM!

**Comment by n-patiphon** [comment]
Thank you everyone for your comments and suggestions! Definitely looks a lot better than the initial version.

**Pull request merged**
